Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -1,6 +1,95 @@
 #include "Training.h"
+#include <stdio.h>
 
 void trainingTask(uint8_t* data)
 {
+    uint8_t validity = 0b00000111;
 
+    checkmotors(&validity, data[0], data[1]);
+
+    checklights(&validity, data[2]);
+
+
+    if( getbit(validity, 0) && getbit(validity, 1)) {
+        outputArray[0] = data[0];
+        outputArray[1] = data[1];
+    }
+    if (getbit(validity, 2)){
+        outputArray[2] = data[2];
+    }
+
+    validData = validity;
+
+}
+
+uint8_t getbit(uint8_t data,int number)
+{
+    return (data & 1<<number);
+}
+
+int getvelocity(uint8_t data)
+{
+    int VelocityBits  = getbit(data, 1) + getbit(data, 2) + getbit(data, 3) + getbit(data, 4) + getbit(data, 5);
+    int velocity = VelocityBits >> 1;
+
+    if (getbit(data, 6))
+    velocity *=-1;
+
+    return velocity;
+}
+
+void checkmotors(uint8_t *validity, uint8_t motor1, uint8_t motor2)
+{
+    int velocity1 = getvelocity(motor1);
+    int velocity2 = getvelocity(motor2);
+
+    int Motor1State = getbit(motor1, 7);
+    if (Motor1State != getbit(motor2, 7)){  //check both motors are either turned on or off
+        *validity &= ~0b00000011;
+        return;
+    } 
+
+    if(Motor1State == 0){ //if motors are off, check velocity is 0
+
+        if (velocity1 != 0)
```

Comment:
Make sure that your spacing and styling are consistent. For example, the spacing between the `if` and the `()`.